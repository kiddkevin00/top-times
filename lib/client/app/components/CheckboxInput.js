import { FaCheck as CheckedMark } from 'react-icons/fa/';
import React, { Component } from 'react';
import cx from 'classnames';
import PropTypes from 'prop-types';

import '../../static/assets/styles/components/checkboxInput.scss';

export default class CheckboxInput extends Component {
  static propTypes = {
    children: PropTypes.node.isRequired,
    onChange: PropTypes.func.isRequired,
    value: PropTypes.bool.isRequired,
    className: PropTypes.string,
  };

  static defaultProps = {
    className: '',
  };

  onKeyPress = event => {
    if (event.key === 'Enter') {
      this.props.onChange(!this.props.value);
    }
  };

  render() {
    const { className, onChange, value, children } = this.props;
    const clazzName = cx('pn-checkbox', className);

    return (
      <div
        className={clazzName}
        onClick={() => onChange(!value)}
        role="button"
        tabIndex="-1"
        onKeyPress={this.onKeyPress}
      >
        <div className="checked-mark-container">
          {value ? <CheckedMark /> : null}
        </div>

        <p>{children}</p>
      </div>
    );
  }
}
