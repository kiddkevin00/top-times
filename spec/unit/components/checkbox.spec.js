import CheckboxInput from '../../../lib/client/app/components/CheckboxInput';
import { shallow, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import React from 'react';

describe('Checkbox Input component', () => {
  describe('when first mounts', () => {
    it('should be rendered correctly', () => {
      const wrapper = shallow(
        <CheckboxInput className="test-class" onChange={() => {}} value={false}>
          This checkbox is for testing
        </CheckboxInput>
      );

      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should always have predefined CSS classes', () => {
      const wrapper = shallow(
        <CheckboxInput className="test-class" onChange={() => {}} value={false}>
          This checkbox is for testing
        </CheckboxInput>
      );

      expect(wrapper.find('.pn-checkbox').length).toBe(1);
    });

    it('should have the correct text rendered', () => {
      const text = 'Click here to agree our terms and services';
      const wrapper = render(
        <CheckboxInput onChange={() => {}} value={false}>
          {text}
        </CheckboxInput>
      );

      expect(wrapper.find('p').text()).toBe(text);
    });
  });

  it('should toggle the checked state when clicking', () => {
    const onChange = jest.fn();
    const wrapper = shallow(
      <CheckboxInput onChange={onChange} value={false}>
        This checkbox is for testing
      </CheckboxInput>
    );

    wrapper.simulate('click');

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(true);
  });

  it('should toggle the checked state when pressing enter', () => {
    const onChange = jest.fn();

    const wrapper = shallow(
      <CheckboxInput onChange={onChange} value={false}>
        This checkbox is for testing
      </CheckboxInput>
    );

    wrapper.simulate('keypress', { key: 'Enter' });

    expect(onChange).toHaveBeenCalledTimes(1);
    expect(onChange).toHaveBeenCalledWith(true);
  });

  it('should not toggle the checked state when pressing other keys', () => {
    const onChange = jest.fn();

    const wrapper = shallow(
      <CheckboxInput onChange={onChange} value={false}>
        This checkbox is for testing
      </CheckboxInput>
    );

    wrapper.simulate('keypress', { key: 'E' });

    expect(onChange).toHaveBeenCalledTimes(0);
  });
});
