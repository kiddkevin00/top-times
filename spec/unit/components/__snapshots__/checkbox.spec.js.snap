// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Input component when first mounts should be rendered correctly 1`] = `
<div
  className="pn-checkbox test-class"
  onClick={[Function]}
  onKeyPress={[Function]}
  role="button"
  tabIndex="-1"
>
  <div
    className="checked-mark-container"
  />
  <p>
    This checkbox is for testing
  </p>
</div>
`;
